src/main/safetydroneflip/safetydroneflip.c

```diff
@@ -0,0 +1,174 @@
+#include "safetydroneflip/safetydroneflip.h"
+
+#include "drivers/time.h"
+
+#include "fc/core.h"
+#include "fc/rc.h"
+#include "fc/rc_controls.h"
+#include "fc/rc_modes.h"
+#include "fc/runtime_config.h"
+
+#include "flight/mixer.h"
+#include "flight/pid.h"
+#include "flight/imu.h"
+
+#include "msp/msp.h"
+
+#include "sensors/battery.h"
+#include "sensors/gyro.h"
+#include "sensors/gyro_init.h"
+#include "sensors/boardalignment.h"
+
+#include "io/motors.h"
+
+#include "build/debug.h"
+
+//check the battery
+
+#define UnsafeValue 430 //unsafe value for the battery normally 430// meervoud van 1 = 0.01 v
+#define UnsafeValueToHigh 480 //unsafe value for the battery normally 430// meervoud van 1 = 0.01 v
+
+int BATTERYCRITICAL= 0;
+
+
+void Motors_out(void) //tuns off motors when neccesary
+{
+    uint8_t motorCount = getMotorCount();
+
+    for (int i = 0; i < motorCount; i++)
+    {
+        motor[i] = 0;
+
+    }
+
+}
+
+void FlipDrone(int Motorthrotle) //controlls the motors
+{ 
+    uint8_t motorCount = getMotorCount();
+
+    for (int i = 0; i < motorCount; i++)
+    {
+        motor[i] = 0;
+    }
+    //led is the back
+    motor[0] = Motorthrotle; //right back
+    motor[1] = 0;   //right front
+    motor[2] = Motorthrotle;   //left back
+    motor[3] = 0;   //left front
+    
+
```

Comment:
During the presentation we heard that it would be very helpful if the drone would stay on the pad during the flip. Luckily, I found a way to reverse the props. You have to call this code once before you turn on the motors and then the two front motors will reverse direction:

```
dshotCommandWrite(1, getMotorCount(), DSHOT_CMD_SPIN_DIRECTION_REVERSED, DSHOT_CMD_TYPE_INLINE);
dshotCommandWrite(3, getMotorCount(), DSHOT_CMD_SPIN_DIRECTION_REVERSED, DSHOT_CMD_TYPE_INLINE);
````
I also think it might help to do a short burst of high thrust.